src/Provider/AbstractProvider.php

```diff
@@ -707,7 +707,7 @@ protected function parseResponse(ResponseInterface $response)
      */
     protected function prepareAccessTokenResponse(array $result)
     {
-        if ($this->getAccessTokenResourceOwnerId()) {
+        if ($this->getAccessTokenResourceOwnerId() !== null) {
```

Comment:
> I'm not sure how Scrutinizer thinks the function returns string|null

Maybe because the default value is `null`? ie. 

`const ACCESS_TOKEN_RESOURCE_OWNER_ID = null;`

The rationalisation behind this change is that if the method doesn't return null, that means the provider explicitly specified *some* value.